extensions/SharkPool/Tune-Shark-V3.js

```diff
@@ -0,0 +1,1627 @@
+  ];
+  const complexEffects = [
+    { text: Scratch.translate("reverb"), value: "reverb" },
+    { text: Scratch.translate("delay"), value: "delay" },
+    { text: Scratch.translate("tremolo"), value: "tremolo" },
+    { text: Scratch.translate("fuzz"), value: "fuzz" },
+    { text: Scratch.translate("bitcrush"), value: "bitcrush" },
+    { text: Scratch.translate("highpass"), value: "highpass" },
+    { text: Scratch.translate("lowpass"), value: "lowpass" },
+    { text: Scratch.translate("flanger"), value: "flanger" },
+    { text: Scratch.translate("compressor"), value: "compressor" },
+    { text: Scratch.translate("equalizer"), value: "equalizer" },
+  ];
+
+  let deltaTime = 0,
+    prevFrameTime = 0;
+  let soundBank = {};
+  let settings = { flagCtrl: false, canSave: false };
+
+  class SPtuneShark3 {
+    constructor() {
+      this.loadStorage = function (storage) {
+        if (storage === undefined) return;
+        settings = storage.settings;
+        soundBank = storage.bank;
+        for (const item in soundBank) {
+          const sound = soundBank[item];
+          sound.loaded = false;
+          if (sound.isVanilla) {
+            const info = sound.src
+              .substring(1, sound.src.lastIndexOf("."))
+              .split("/");
+            let target =
+              info[0] === "Stage"
+                ? runtime.getTargetForStage()
+                : runtime.getSpriteTargetByName(info[0]);
+            if (target === undefined) {
+              alert(
+                Scratch.translate(
+                  `Tune Shark 3 -- Failed to load ${info[1]} from ${info[0]}`
+                )
+              );
+              continue;
+            }
+
+            const scratchSound = target.sprite.sounds.find((i) => {
+              return i.name === info[1];
+            });
+            if (scratchSound === undefined) {
+              alert(
+                Scratch.translate(
+                  `Tune Shark 3 -- Failed to load ${info[1]} from ${info[0]}`
+                )
+              );
+              continue;
+            }
+            const buffer =
+              target.sprite.soundBank.soundPlayers[scratchSound.soundId].buffer;
+            const engine = new Pizzicato.Sound({
+              source: "buffer",
+              options: { buffer, attack: 0 },
+            });
+
+            engine.sourceNode = engine.getSourceNode();
+            sound.context = engine;
+            sound.loaded = true;
+          } else {
+            const engine = new Pizzicato.Sound(
+              {
+                source: "file",
+                options: { path: sound.src, attack: 0 },
+              },
+              () => {
+                engine.sourceNode = engine.getSourceNode();
+                sound.context = engine;
+                sound.loaded = true;
+              }
+            );
+          }
+        }
+      };
+      if (!Scratch.extensions.isPenguinMod) {
```

Comment:
This is the TurboWarp repository, we rather you don't add checks for things like PenguinMod and whatnot or using PenguinMod APIs.